superset/views/database/forms.py

```diff
@@ -303,8 +300,11 @@ def at_least_one_schema_is_allowed(database: Database) -> bool:
         ],
     )
 
-    sheet_name = StringField(
-        _("Sheet Name"), description="Sheet Name", validators=[Optional()]
+    sheet_name = IntegerField(
+        _("Sheet Name"),
+        description="Integers used in zero-indexed sheet positions (0 is first sheet).",
+        validators=[Optional(), NumberRange(min=0)],
+        widget=BS3TextFieldWidget(),
     )
```

Comment:
The `sheet_name` could be a type of string

Reply:
the default value for sheet_name is 0, see https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_excel.html, and if it is set to string with default value None, all sheets will be read, and the result will be a dict of dataframes, which is not desired and cause exception, so I use integer instead.

Reply:
Base on the **pandas** docs the sheet_name could be `string`, `int`, `list` and `None`. The `StringField` is better than `IntegerField` because of Softwares that parse excel format like **Microsoft office** and **LibreOffice** save the sheets name as `string` and by default, it is `Sheet1`. In addition, the sheet_name want the name of the sheet!. In the case of the number like 0,1, etc it would be `sheet_number`.
I totally agree to change the default value from `None` to the `0`.
thanks